### DISCUSSION TITLE
Ability to ignore triggers on a query

### DISCUSSION THREAD

**Discussion by Unskilledcrab** [koenbeuk/EntityFrameworkCore.Triggered#113] [Ideas]
**Issue**
I have a part of my system that is constantly hitting the API and calling an update on a table, I do not want this specific call to pass through the trigger pipeline.

**The current work around**
This performance hit can be avoided by using a different ORM for the query or by running the query as raw sql

**Goal**
It would be nice to have an extension method to add to a ef query that would be similar to ef cores AsNoTracking() _maybe WithNoTriggers()_ that would allow the user to make sure that triggers would not be called during the pipeline to improve performance for queries

**Comment by koenbeuk** [comment]
This is an interesting idea. 

What you're asking for is a way of 'marking' entities to not raise triggers for, so given the following example:

```csharp
var jon = dbContext.Users.WithNoTriggers().FirstOrDefault(x => x.Name == "Jon");
var jane = dbContext.Users.FirstOrDefault(x => x.Name == "Jane");
jon.Counter += 1;
jane.Counter += 1;
dbContext.SaveChanges();
```

This will only raise triggers for Jane as Jon has been tracked with a hypothetical WithNoTriggers behavior. This will still require a design in order to mark new and attached entities to not raise triggers.

I've thought about something similar (yet different), namely the ability to disable triggers on demand. Something like the following:

```csharp
dbContext.GetTriggerService().DisableTriggers();
dbContext.SaveChanges(); // This will NOT raise any triggers
dbContext.GetTriggerService().EnableTriggers();

// Or in short an extension method: dbContext.SaveChangesWithNoTriggers();
```

This does in turn not allow the fine grained control to only enable Triggers for specific entities but It would simplify a whole lot.

Would my alternative approach work in your case?

**Reply by Unskilledcrab** [reply]
Yes I believe a global disable of the triggers with the on save event would definitely work! I think if that is the quickest option to implement then it would be great!

**Comment by koenbeuk** [comment]
This is tracked in #114

**Comment by koenbeuk** [comment]
Support for this is now in the master branch and will soon be released (under preview).

The following can be used to save changes without invoking triggers:
```csharp
using EntityFrameworkCore.Triggered.Extensions;
dbContext.SaveChangesWithoutTriggers();
```

**Reply by Unskilledcrab** [reply]
Is this method async?

**Reply by Unskilledcrab** [reply]
Ignore that,  just looked through the commit and saw the async version. Thank you!!!
